exonum-java-binding-common/src/main/java/com/exonum/binding/common/proofs/list/UncheckedListProofAdapter.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2018 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.common.proofs.list;
+
+import com.exonum.binding.common.hash.HashCode;
+import com.exonum.binding.common.proofs.common.ProofStatus;
+import com.exonum.binding.common.serialization.Serializer;
+import com.google.common.base.Preconditions;
+
+/**
+ * An Adapter class used to simplify work with ListProof interfaces.
+ */
+public class UncheckedListProofAdapter<E> implements UncheckedListProof {
+
+  private final ListProofStructureValidator listProofStructureValidator;
```

Comment:
I believe we can now hide the structure validator and root hash calculator
inside the package and use this class outside:
  - As the return value of `ProofList#getProof` & `#getrangeproof`
  - As the thing we use in the client code (e.g., proof matchers).

Reply:
Agree. That was the main idea of this refactoring.